### PR TITLE
Milestone #3: Set Up Associations

### PR DESCRIPTION
**Pull request by Anjacodes**
In this milestone, I:

* Created the has-many/belongs-to relationship between the newly created class `Classroom` and `Student`.
* Created the many-to-many relationship between `Person` and the newly created `Book` class using the newly created intermediate class `Rental`
* Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
* Made sure to use correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
* Made sure to document my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)[](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#ruby-requirements).
* Followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @age = age
     @name = name
     @parent_permission = parent_permission
     @name_corrector = Corrector.new
+    @rentals = []
```
Comment (mmsesay) [inline person.rb:13]: - [ ] A method for adding new rentals for a person should be present here just after the initialize method.

See example:
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (mmsesay) [inline book.rb]: - [ ] Add method for adding new rentals for a book should be present here just after the initialize method. This function will help save records about rented books and who rents them.

See example:
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
end
```

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @Anjacodes, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Anjacodes) [reply person.rb:13]: Hi @mmsesay, thank you for the feedback, I had totally overlooked that 🙏! 
I have implemented the suggested changes and pushed again. Appreciate your time!

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @Anjacodes,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**